controllers/C_Prescription.class.php

```diff
@@ -40,6 +40,54 @@ function __construct($template_mod = "general")
         $this->pconfig = $GLOBALS['oer_config']['prescriptions'];
         $this->RxList = new RxList();
 
+        $interaction = "";
+
+        /*
+        *   check to see if RxNorm installed
+        */
+        $rxn = sqlQuery("SELECT table_name FROM information_schema.tables WHERE table_name = 'RXNCONSO' OR table_name = 'rxnconso'");
+        if ($rxn == false) {
+           $interaction = "Could not find RxNorm Table! Please install.";
+        } elseif ($rxn == true) {
+           /*
+            *   Grab medication list from prescriptions list
+            *   load into array
+            */
+           $pid = $GLOBALS['pid'];
+            $medList = sqlStatement("SELECT drug FROM prescriptions WHERE active = 1 AND patient_id = ?", array($pid));
+            $nameList = array();
+            while ($name = sqlFetchArray($medList)) {
+                $drug = explode(" ", $name['drug']);
+                $rXn = sqlQuery("SELECT `rxcui` FROM `" . mitigateSqlTableUpperCase('RXNCONSO') . "` WHERE `str` LIKE ?", array("%" . $drug[0] . "%"));
+                $nameList[] = $rXn['rxcui'];
+            }
+            if (count($nameList) < 2) {
+                $interaction = "Need more than one drug.";
+
+            } else {
+                /*
+                *  If there are drugs to compare, collect the data
+                */
+                $rxcui_list = implode(", ", $nameList);
+                $data = file_get_contents("https://rxnav.nlm.nih.gov/REST/interaction/list.json?rxcuis=".$rxcui_list);
```

Comment:
use function to encode parameters in the url (stops special chars from breaking anything):
```
$data = file_get_contents("https://rxnav.nlm.nih.gov/REST/interaction/list.json?rxcuis=" . urlencode($rxcui_list));
```